src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala

```diff
@@ -49,6 +61,21 @@ class ChargingFunctions[GEO: GeoLevel](
     }
   }
 
+  /**
+    * function that verifies if Enroute Then Fast Charging Only
+    * @param zone ParkingZone
+    * @param inquiry ParkingInquiry
+    * @return
+    */
+  def ifEnrouteThenFastChargingOnly(zone: ParkingZone[GEO], inquiry: ParkingInquiry): Boolean = {
```

Comment:
Yes, but it's standalone method and name is important here. It doesn't just check the charging point but the search mode has to be of the type enroute.

Reply:
I agree the name matters - that is why it confuses me - as it returns true if it is not enroute, thus muddling the name with the implementation. This could lead to future code usage based on the name which is not accurate. As such, I suggest one of two options:

# Rename it to `isEnrouteFastChargingOnlyOrNotEnroute`
# Push the extra implementation out

```
  def isEnrouteAndFastChargingOnly(zone: ParkingZone[GEO], inquiry: ParkingInquiry): (Boolean, Option[Boolean]) = {
    inquiry.searchMode match {
      case ParkingSearchMode.EnRoute =>
        (true, Some(ChargingPointType.isFastCharger(zone.chargingPointType.get))
      case _ => (false, None)
    }
  }
```

Then in the usage:

```
  ...
  val (isEnroute, maybeIsFastChargingOnly): Boolean = ifEnrouteThenFastChargingOnly(zone, inquiry)
  ...
  isEV && rideHailFastChargingOnly && validChargingCapability && canCarParkHere && 
    (!isEnroute || (isEnroute && maybeIsFastChargingOnly.getOrElse(false)))
```

Honestly, for the limited usage, I would rename the method and variable personally :) But it depends on how ?extractionary? you want to be.